src/main/java/uk/co/onsdigital/discovery/metadata/api/service/UrlBuilder.java

```diff
@@ -0,0 +1,124 @@
+package uk.co.onsdigital.discovery.metadata.api.service;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriTemplate;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Specialised URL builder service to centralise construction of external URLs.
+ */
+@Service
+public class UrlBuilder {
+    static final int MAX_PAGE_SIZE = 1000;
+
+    private final String baseUrl;
+
+    private final UriTemplate pageTemplate;
+    private final UriTemplate dataSetTemplate;
+    private final UriTemplate dimensionalDataSetTemplate;
+    private final UriTemplate dimensionsTemplate;
+    private final UriTemplate dimensionTemplate;
+    private final UriTemplate hierarchyTemplate;
+
+    UrlBuilder(@Value("#{systemEnvironment['BASE_URL'] ?: 'http://localhost:20099'}") String baseUrl) {
+        this.baseUrl = requireNonNull(baseUrl);
+
+        pageTemplate = new UriTemplate(baseUrl + "/datasets?page={page}&size={size}");
+        dataSetTemplate = new UriTemplate(baseUrl + "/datasets/{dataSetId}");
+        dimensionalDataSetTemplate = new UriTemplate(baseUrl + "/datasets/{dataSetId}/editions/{edition}/versions/{version}");
+        dimensionsTemplate = new UriTemplate(baseUrl + "/datasets/{dataSetId}/editions/{edition}/versions/{version}/dimensions");
+        dimensionTemplate = new UriTemplate(baseUrl + "/datasets/{dataSetId}/editions/{edition}/versions/{version}/dimensions/{dimensionId}");
+        hierarchyTemplate = new UriTemplate(baseUrl + "/hierarchies/{hierarchyId}");
+    }
+
+    /**
+     * Returns a template function that can be used to construct URLs for navigating to particular pages.
+     *
+     * @param pageSize the number of items on each page.
+     * @return a {@link PageUrlTemplate} for building links to individual pages in the result set.
+     */
+    public PageUrlTemplate datasetsPage(int pageSize) {
+        if (pageSize < 1) {
+            throw new IllegalArgumentException("pageSize should be >= 1");
+        }
+        if (pageSize > MAX_PAGE_SIZE) {
+            throw new IllegalArgumentException("pageSize must be <= 1000");
```

Comment:
Will do, this was part of a copy pasta. I'll change it in both this and the `LegacyUrlBuilder`.